### DISCUSSION TITLE
Sync vs Async

### DISCUSSION THREAD

**Discussion by AmritaDatta** [webdriverio/webdriverio#8180] [Q&A]
Hi Experts,
I've a query: I'm trying to write the following code in asynchronous mode?

        const chai = require("chai");
        static table() { return $$('tbody[id$="idBgTable-tblBody"] tr[id$="idTable"]'); }
        static noDataRows() {
        const noDataRows = BsgGroup.table();
        noDataRows[0].waitForDisplayed();
        chai.expect(noDataRows.length === 1).to.be.true; }

My approach in page.js:
        
        const chai = require("chai");
        static table() { return $$('tbody[id$="idBgTable-tblBody"] tr[id$="idTable"]'); }   --> I've doubt here, should I write async / await here too??
        static async noDataRows() {
        const noDataRows = await BsgGroup.table();
        await noDataRows[0].waitForDisplayed();
        chai.expect(noDataRows.length === 1).to.be.true; }  --> I've doubt here, should I write await here too??

Could anyone please let me know if it's correct or not??? [Although test is passed in node v.16 and without @wdio/sync] Thank you in advance :)

**Comment by christian-bromann** [comment]
Hey @AmritaDatta , your code looks about right!

**Reply by AmritaDatta** [reply]
Hi @christian-bromann Thank you so much for your reply. That means my last line: chai.expect(noDataRows.length === 1).to.be.true; } is correct. We don't need to use await befor "chai.expect".

**Reply by christian-bromann** [reply]
no, nothing in this line is a promise

**Reply by AmritaDatta** [reply]
Could you please take a look at it and let me know if it's correct or not? Thank you and sorry if I bother you.
When(/^I wait for (.*)ms$/, async(iDelay)  => {
	const starttime = new Date().getTime();
    await browser.pause(iDelay);
    const endtime = new Date().getTime();
    console.log("Navigation - Slept for " + (endtime - starttime) + "ms");
});

**Reply by christian-bromann** [reply]
this is correct
